### ISSUE TITLE
Start Redis lifecycle tests sooner

### ISSUE THREAD

**Issue by krancour** [Azure/open-service-broker-azure#148]
Our lifecycle tests run up to ten test cases in parallel. (If we do more than that, we aren't likely to be able to do CI on multiple PRs in parallel without bumping into various quotas on our CI/CD Azure subscription.)

If I am counting correctly, Redis (which takes the longest to provision) is currently the tenth test case-- meaning it's just barely sneaking into the first wave. The test cases are currently ordered alphabetically by module. Addition of new modules (or additional test cases for existing modules) will bump Redis out of the first wave of tests and this will certainly have a deleterious effect on the total runtime of the lifecycle tests.

The Redis test cases should be kicked off _first_ so we're launching these long-running tests earlier, rather than on the tail end of the process.

**Comment by zhongyi-zhang** [comment]
I think we can solve it for not only Redis case but also more and more cases we will have in future (from new modules and existing modules with new features).
We can add an *expected period* to each case.
  * Manually assign a value to initialize it according to a test result.
  * Sort the cases by this property in descending order at https://github.com/Azure/open-service-broker-azure/blob/master/tests/lifecycle/test_cases_test.go#L41
  * Log a warning if the period difference between *expected* and *actual* is too big (>20% maybe) in some case. Then we can be aware of it and update the value of this property.

The approach should work well if these Azure services are stable enough in time scale.

**Comment by krancour** [comment]
If we manually assign an expected execution time to each test case, it's not too much different from manually arranging the test cases in order with those that take longest starting first.

I just want to do the simplest thing that could work here and not get more sophisticated until there's a demonstrated need for it.

_Timeouts_ for these tests wouldn't be a bad idea. That's something that could that could be explored as its own issue / PR.

**Comment by zhongyi-zhang** [comment]
> If we manually assign an expected execution time to each test case, it's not too much different from manually arranging the test cases in order with those that take longest starting first.

I don't think they are not too much different... With more long-execution-time modules checking in, we wouldn't remember which one takes longer time. And the approach I said could arrange the test cases by case, while manually arranging is by module.
But I agree with you it's too sophisticated for now.

If we just want the simplest thing, could we separate the modules into two groups, one for long-execution-time modules and the other one for other modules, to push into `getTestCaseFuncs`? https://github.com/Azure/open-service-broker-azure/blob/master/tests/lifecycle/test_cases_test.go#L17-L32
